src/config.ts

```diff
@@ -451,19 +459,30 @@ function loadPlugins(
   // if no mounted directories, mount root
   if (!Object.keys(mountedDirs).length) mountedDirs['.'] = '/';
 
-  const needsDefaultPlugin = new Set(['.js', '.mjs', '.jsx', '.ts', '.tsx']);
+  const needsDefaultPlugin = new Set(['.mjs', '.jsx', '.ts', '.tsx']);
   plugins
-    .reduce((arr, a) => arr.concat(a.input), [] as string[])
+    .filter(({resolve}) => !!resolve)
+    .reduce((arr, a) => arr.concat(a.resolve!.input), [] as string[])
     .forEach((ext) => needsDefaultPlugin.delete(ext));
   if (needsDefaultPlugin.size > 0) {
     plugins.unshift(esbuildPlugin({input: [...needsDefaultPlugin]}));
   }
 
+  const extensionMap = plugins.reduce((map, {resolve}) => {
```

Comment:
Also, I believe you can pass the generic into the `reduce` function instead of casting the initial value.

```suggestion
  const extensionMap = plugins.reduce<Record<string, string>>((map, {resolve}) => {
```